From 1e33ee476f377d0f193091ef6ae921c347835be6 Mon Sep 17 00:00:00 2001
From: Stefan Hansson <newbyte@postmarketos.org>
Date: Mon, 20 Oct 2025 10:02:07 +0200
Subject: [PATCH] conf-tweaks: Remove nonsensical assertion

I'm not sure why I added this assertion, but my guess would be that
either `from` also used to be a pointer, or I just didn't really think
about that it wasn't one when adding that. Either way, it makes no sense
since it's perfectly valid for the weigh of a setting to be 0. Asserting
that `into` is not 0 makes sense since it is a pointer and this function
will crash if `into` is a NULL pointer.

Closes https://gitlab.gnome.org/World/Phosh/phosh-mobile-settings/-/issues/88

Fixes b113b068e26ce11cb645b5570d9f00a999cb8de0

Signed-off-by: Stefan Hansson <newbyte@postmarketos.org>
Part-of: <https://gitlab.gnome.org/World/Phosh/phosh-mobile-settings/-/merge_requests/271>
---
 src/conf-tweaks/ms-tweaks-parser.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/conf-tweaks/ms-tweaks-parser.c b/src/conf-tweaks/ms-tweaks-parser.c
index e01f02c4..79d02199 100644
--- a/src/conf-tweaks/ms-tweaks-parser.c
+++ b/src/conf-tweaks/ms-tweaks-parser.c
@@ -421,7 +421,6 @@ static void
 merge_weights (int *into, const int from)
 {
   g_assert (into);
-  g_assert (from);
 
   /* Only overwrite the weight if it wasn't specified. We assume that even if the default was
    * specified, the original definition didn't care about the weight. */
-- 
GitLab

